api/v1alpha1/hostedcluster_types.go

```diff
@@ -740,6 +740,72 @@ type KubevirtPlatformSpec struct {
 	// the same cluster and namespace as the Hosted Control Plane.
 	// +optional
 	Credentials *KubevirtPlatformCredentials `json:"credentials,omitempty"`
+
+	// StorageDriver defines how the KubeVirt CSI driver exposes StorageClasses on
+	// the infra cluster (hosting the VMs) to the guest cluster.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	// +immutable
+	StorageDriver *KubevirtStorageDriverSpec `json:"storageDriver,omitempty"`
+}
+
+// KubevirtStorageDriverConfigType defines how the kubevirt storage driver is configured.
+//
+// +kubebuilder:validation:Enum=None;Default;Manual
+type KubevirtStorageDriverConfigType string
+
+const (
+	// NoneKubevirtStorageDriverConfigType means no kubevirt storage driver is used
+	NoneKubevirtStorageDriverConfigType KubevirtStorageDriverConfigType = "None"
+
+	// DefaultKubevirtStorageDriverConfigType means the kubevirt storage driver maps to the
+	// underlying infra cluster's default storageclass
+	DefaultKubevirtStorageDriverConfigType KubevirtStorageDriverConfigType = "Default"
+
+	// UserKubevirtStorageDriverConfigType means the kubevirt storage driver mapping is
+	// explicitly defined.
+	UserKubevirtStorageDriverConfigType KubevirtStorageDriverConfigType = "Manual"
+)
+
+type KubevirtStorageDriverSpec struct {
+	// Type represents the type of kubevirt csi driver configuration to use
+	//
+	// +unionDiscriminator
+	// +immutable
+	// +kubebuilder:default=Default
+	// +optional
+	Type KubevirtStorageDriverConfigType `json:"type,omitempty"`
+
+	// Manual is used to explicilty define how the infra storageclasses are
+	// mapped to guest storageclasses
+	//
+	// +immutable
+	// +optional
+	Manual *KubevirtManualStorageClassMapping `json:"manual,omitempty"`
+}
+
+type KubevirtManualStorageClassMapping struct {
```

Comment:
Maybe `KubevirtStorageDriverConfig`?  Especially if there may eventually be other stuff in here.

Reply:
yep, fixed.